example/lib/helpers/local_realm_provider.dart

```diff
@@ -39,3 +45,88 @@ class _RealmProviderState extends State<RealmProvider> {
     );
   }
 }
+
+class _InMemoryRealmProviderState extends _RealmProviderState {
+
+  @override
+  void initState() {
+    super.initState();
+    final configuration = Configuration(inMemoryIdentifier: Uuid().v4());
+    realm = Realm.open(configuration);
+  }
+
+}
+
+class _FileRealmProviderState extends _RealmProviderState {
+
+  @override
+  void initState() {
+    super.initState();
+    final configuration = Configuration();
+    realm = Realm.open(configuration);
+  }
+
+}
+
+class _FileNameRealmProviderState extends _RealmProviderState {
+
+  @override
+  void initState() {
+    super.initState();
+    final configuration = Configuration(
+      fileName: 'custom_name.realm'
+    );
+    realm = Realm.open(configuration);
+  }
+
+}
+
+class _FileDirectoryRealmProviderState extends _RealmProviderState {
+
+  @override
+  void initState() {
+    super.initState();
+    final configuration = Configuration(
+      fileDirectory:  'custom_dir/'
+    );
+    realm = Realm.open(configuration);
+  }
+
+}
+
+class _FileDirectoryNameRealmProviderState extends _RealmProviderState {
+
+  @override
+  void initState() {
+    super.initState();
+    final configuration = Configuration(
+      fileDirectory: 'custom_dir/sub_dir/',
+      fileName: 'custom_name.realm'
+    );
+    realm = Realm.open(configuration);
+  }
+
+}
+
+class _EncryptedFileRealmProviderState extends _RealmProviderState {
+
+  @override
+  void initState() {
+    super.initState();
+    final encryptionKey = _secureRandom(64);
```

Comment:
Note that the example generates a new encryption key every time the example is run. You will get a fatal error if you run the example a 2nd time because the keys will not match. It might be better to write the 1st encryption key to a file for demo purposes so someone can re-open the encrypted Realm.

Reply:
Why don't use constant in demo?

Reply:
Great idea. Will update.